### PR TITLE
[WIP] Added automatic port info detection from Dockerfile

### PR DESCRIPTION
**Pull request by surajnarwade**
Experimental

### CODE REVIEW

File: cmd/init.go [hunk 1]
```diff
@@ -84,6 +85,25 @@ var initCmd = &cobra.Command{
 
 		if len(ports) > 0 {
 			obj.Services = []Service{{PortMappings: ports}}
+		} else {
+			// Dockerfile detection
+			_, err = os.Stat("Dockerfile")
```
Comment (surajssd) [inline cmd/init.go]: this will pickup dockerfile relative to kedge binary not the project root dir

Comment (surajssd) [reply cmd/init.go]: parameterize the dockerfile name to a variable and use it everwhere

File: cmd/init.go [hunk 2]
```diff
@@ -84,6 +85,25 @@ var initCmd = &cobra.Command{
...
+			if err != nil {
+				fmt.Println(fileName, "is not present")
```
Comment (surajssd) [inline cmd/init.go]: i am not sure what fileName is here?

File: cmd/init.go [hunk 3]
```diff
@@ -21,6 +21,7 @@ import (
 	"os"
 	"strings"
 
+	"github.com/asottile/dockerfile"
```
Comment (surajssd) [inline cmd/init.go]: Q: no need to here?

File: cmd/init.go [hunk 4]
```diff
@@ -84,6 +85,25 @@ var initCmd = &cobra.Command{
...
 		if len(ports) > 0 {
 			obj.Services = []Service{{PortMappings: ports}}
+		} else {
+			// Dockerfile detection
+			_, err = os.Stat("Dockerfile")
+			if err != nil {
+				fmt.Println(fileName, "is not present")
+				os.Exit(-1)
+			} else {
+				a, err := dockerfile.ParseFile("Dockerfile")
```
Comment (surajssd) [inline cmd/init.go]: the parsing should remain generic, as in whether to parse or not should be governed by other factors(like a flag or something) not on whether port info is provided.

Comment (surajssd) [reply cmd/init.go]: in this section then only extract port info

**Review by surajssd — CHANGES_REQUESTED** [review]

File: cmd/init.go (hunk 3 above)
Comment (surajnarwade) [reply cmd/init.go]: done vendoring

**Comment by surajnarwade** [comment]
There is conflict between docker version of `dockerfile` and `kedge` :(

**Pull request closed without merging**